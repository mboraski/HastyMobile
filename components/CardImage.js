import React from 'react';
import { Image, StyleSheet } from 'react-native';

import bankAmericaIcon from '../assets/icons/bank-america.png';
import masterCardIcon from '../assets/icons/master-card.png';
import visaIcon from '../assets/icons/visa.png';
import paypalIcon from '../assets/icons/paypal.png';

const VISA = 'visa';
const MASTERCARD = 'master-card';
const AMERICAN_EXPRESS = 'american-express';
const DINERS_CLUB = 'diners-club';
const DISCOVER = 'discover';
const JCB = 'jcb';
const UNIONPAY = 'unionpay';
const MAESTRO = 'maestro';
// not supported
const BANK_AMERICA = 'bank-america';

const sources = {
    [BANK_AMERICA]: bankAmericaIcon,
    [MASTERCARD]: masterCardIcon,
    // [AMERICAN_EXPRESS]: americanExpressIcon,
    // [DINERS_CLUB]: dinersClubIcon,
    // [DISCOVER]: discoverIcon,
    // [JCB]: jcbIcon,
    // [UNIONPAY]: unionpayIcon,
    // [MAESTRO]: maestroIcon,
    [VISA]: visaIcon,
    // discover: discoverIcon,
    // maestro: maestroIcon,
    paypal: paypalIcon
};

const CardImage = ({ type, style, ...props }) => (
    <Image {...props} source={sources[type]} style={[styles.image, style]} resizeMode="contain" />
);

const styles = StyleSheet.create({
    image: {
        alignSelf: 'center',
        width: 50
    }
});

export default CardImage;
