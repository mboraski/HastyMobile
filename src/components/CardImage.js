import React from 'react';
import { Image, StyleSheet } from 'react-native';

import masterCardIcon from '../assets/icons/cc-mastercard.png';
import visaIcon from '../assets/icons/cc-visa.png';
import americanExpressIcon from '../assets/icons/cc-amex.png';
import dinersClubIcon from '../assets/icons/cc-diners.png';
import discoverIcon from '../assets/icons/cc-discover.png';
import jcbIcon from '../assets/icons/cc-jcb.png';
import maestroIcon from '../assets/icons/cc-maestro.png';

const VISA = 'visa';
const MASTERCARD = 'master-card';
const AMERICAN_EXPRESS = 'american-express';
const DINERS_CLUB = 'diners-club';
const DISCOVER = 'discover';
const JCB = 'jcb';
const MAESTRO = 'maestro';

const sources = {
    [MASTERCARD]: masterCardIcon,
    [AMERICAN_EXPRESS]: americanExpressIcon,
    [DINERS_CLUB]: dinersClubIcon,
    [DISCOVER]: discoverIcon,
    [JCB]: jcbIcon,
    [MAESTRO]: maestroIcon,
    [VISA]: visaIcon
};

const CardImage = ({ type, style, ...props }) => (
    <Image
        {...props}
        source={sources[type]}
        style={[styles.image, style]}
        resizeMode="contain"
    />
);

const styles = StyleSheet.create({
    image: {
        alignSelf: 'center',
        width: 50
    }
});

export default CardImage;
