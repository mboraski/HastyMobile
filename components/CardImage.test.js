import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import CardImage from './CardImage';

describe('CardImage', () => {
    it('renders correctly', () => {
        const render = shallow(<CardImage />);
        expect(render).toMatchSnapshot();
    });
});
