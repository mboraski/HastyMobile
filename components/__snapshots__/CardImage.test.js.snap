// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImage renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Image
    resizeMode="contain"
    source={undefined}
    style={
        Array [
            Object {
              "alignSelf": "center",
              "width": 50,
            },
            undefined,
          ]
    }
/>,
  "nodes": Array [
    <Image
      resizeMode="contain"
      source={undefined}
      style={
            Array [
                  Object {
                    "alignSelf": "center",
                    "width": 50,
                  },
                  undefined,
                ]
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <CardImage />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Image
          resizeMode="contain"
          source={undefined}
          style={
                    Array [
                              Object {
                                "alignSelf": "center",
                                "width": 50,
                              },
                              undefined,
                            ]
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <Image
          resizeMode="contain"
          source={undefined}
          style={
                    Array [
                              Object {
                                "alignSelf": "center",
                                "width": 50,
                              },
                              undefined,
                            ]
          }
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CardImage />,
}
`;
